internal/timeline/timeline.go

```diff
@@ -0,0 +1,195 @@
+package timeline
+
+import (
+	"fmt"
+	"log"
+	"sync"
+)
+
+const EPSILON TIME = 0.0001
+const DEFAULT_TRANSITION TIME = 0.5
+
+var debugTimeline bool = false
+var idSeed int64
+var idSeedMutex sync.Mutex
+
+type ON_STEP func(*TIME) ITimeline
```

Comment:
Please refer to the naming conventions of other types in the project.

Reply:
ON_STEP will be removed